src/shutdown/shutdown.c

```diff
@@ -532,6 +532,7 @@ int main(int argc, char *argv[]) {
         }
 
         /* We're done with the watchdog. */
+        watchdog_close(true);
```

Comment:
This looks wrong to me. If anything happens after this point that hangs the system then we want the watchdog to trigger a reset. So disarming the watchdog here is wrong.

Reply:
Then I suppose we should move the whole "We're done with the watchdog" to fair end enough, like:
diff --git a/src/shutdown/shutdown.c b/src/shutdown/shutdown.c
index 15fd6f93d3..c57834ee7c 100644
--- a/src/shutdown/shutdown.c
+++ b/src/shutdown/shutdown.c
@@ -531,10 +531,6 @@ int main(int argc, char *argv[]) {
                           need_md_detach ? " MD devices," : "");
         }
 
-        /* We're done with the watchdog. */
-        watchdog_close(true);
-        watchdog_free_device();
-
         arguments[0] = NULL;
         arguments[1] = arg_verb;
         arguments[2] = NULL;
@@ -629,6 +625,10 @@ int main(int argc, char *argv[]) {
                 assert_not_reached();
         }
 
+        /* We're done with the watchdog. */
+        watchdog_close(true);
+        watchdog_free_device();
+
         (void) reboot(cmd);
         if (errno == EPERM && in_container) {
                 /* If we are in a container, and we lacked

Reply:
Even the reboot can hang if there is an issue with a driver, etc. I think we should just remove `watchdog_close()` and `watchdog_free_device()`. That way, there will be no confusing output and the watchdog remains active.